packages/neso-particles/package.py

```diff
@@ -0,0 +1,38 @@
+from spack import *
+import os
+import shutil
+
+
+class NesoParticles(CMakePackage):
+    """NESO-Particles"""
+
+    homepage = "https://excalibur-neptune.github.io/NESO-Particles/"
+
+    git = "https://github.com/ExCALIBUR-NEPTUNE/NESO-Particles.git"
+
+    version("0.1.0", commit="2e1b1aac6f4f9c22b31787cf300fcc2b4914cbe1", preferred=True)
+    # 22/08/2022
+    version("0.0.1-f54dc8", commit="10a54a410ddd0eb5d1debc702f415dd798eb89ee")
```

Comment:
I'd recommend against including a hash in the version. If people want to build against a specific hash it is possible to request that when [specifying a version](https://spack.readthedocs.io/en/latest/basic_usage.html#version-specifier) for spack to fetch.

Reply:
Specify a tag instead? (spack packaging guide seems to prefer specifying commits here https://spack.readthedocs.io/en/latest/packaging_guide.html). Not specifying a tag/commit seems fragile, i.e. I can add
```
version("0.2.0")
```
to the package and do
```
spack install neso.neso-particles@0.2.0 ^neso.hipsycl ^mpich
```
and spack happily installs something and returns a successful install. Except NESO-Particles doesn't have a 0.2.0 release. So it definitely hasn't installed 0.2.0.....

Reply:
Oh I see, I thought you were referring to line 13 and was confused.

Reply:
Pushed update.